airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -736,23 +741,31 @@ def create_default_paginator(
         config: Config,
         *,
         url_base: str,
+        decoder: Optional[Decoder] = None,
         cursor_used_for_stop_condition: Optional[DeclarativeCursor] = None,
     ) -> Union[DefaultPaginator, PaginatorTestReadDecorator]:
-        decoder = self._create_component_from_model(model=model.decoder, config=config) if model.decoder else JsonDecoder(parameters={})
+        if decoder:
+            decoder_to_use = decoder
+        elif model.decoder:
+            decoder_to_use = self._create_component_from_model(model=model.decoder, config=config)
+        else:
+            decoder_to_use = JsonDecoder(parameters={})
+        if not isinstance(decoder_to_use, JsonDecoder):
+            raise ValueError(f"Provided decoder of {type(decoder_to_use)=} is not supported. Please set JsonDecoder instead.")
```

Comment:
@artem1205 is there a follow up issue to clean up the yaml interface?

Reply:
created https://github.com/airbytehq/airbyte-internal-issues/issues/8760